### PR TITLE
resource/alicloud_cs_kubernetes_node_pool: update sdk and optimize resource update logic

### PR DESCRIPTION
**Pull request by Pangjiping**
#5181

### CODE REVIEW

**Comment by Pangjiping** [comment]
field-check depends on #5056 
@xiaozhu36

**Comment by Pangjiping** [comment]
#5181

File: alicloud/resource_alicloud_cs_kubernetes_node_pool.go [hunk 1]
```diff
@@ -728,60 +729,44 @@ func resourceAlicloudCSNodePoolUpdate(d *schema.ResourceData, meta interface{})
...
 	}
 
-	if v, ok := d.GetOk("scaling_config"); ok && v != nil {
-		args.AutoScaling = setAutoScalingConfig(v.([]interface{}))
-	}
 	if d.HasChange("scaling_config") {
 		update = true
-		if v, ok := d.GetOk("scaling_config"); ok {
-			args.AutoScaling = setAutoScalingConfig(v.([]interface{}))
+		if v, ok := d.Get("scaling_config").([]interface{}); ok {
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_cs_kubernetes_node_pool.go:746]: 直接用 GetOk

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: alicloud/resource_alicloud_cs_kubernetes_node_pool.go [hunk 2]
```diff
@@ -1048,19 +1034,19 @@ func buildNodePoolArgs(d *schema.ResourceData, meta interface{}) (*cs.CreateNode
 	creationArgs := &cs.CreateNodePoolRequest{
 		RegionId: common.Region(client.RegionId),
 		NodePoolInfo: cs.NodePoolInfo{
-			Name:         d.Get("name").(string),
-			NodePoolType: "ess", // hard code the type
+			Name:         tea.String(d.Get("name").(string)),
```
[the remark this answers is not in the document]
Comment (Pangjiping) [reply alicloud/resource_alicloud_cs_kubernetes_node_pool.go]: done. .(string)是多余的

File: alicloud/resource_alicloud_cs_kubernetes_node_pool.go (hunk 1 above)
Comment (Pangjiping) [reply alicloud/resource_alicloud_cs_kubernetes_node_pool.go]: done

**Pull request merged**